packages/details/src/summary-controller.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Copyright (c) 2019 - 2022 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { SlotChildObserveController } from '@vaadin/component-base/src/slot-child-observe-controller.js';
+
+/**
+ * A controller to manage the summary element.
+ */
+export class SummarySlotController extends SlotChildObserveController {
+  constructor(host, tagName, config) {
+    super(host, 'summary', tagName, config);
+  }
+
+  /**
+   * Set summary based on corresponding host property.
+   *
+   * @param {string} summary
+   */
+  setSummary(summary) {
+    this.summary = summary;
+
+    // Restore the default summary, if needed.
+    const summaryNode = this.getSlotChild();
+    if (!summaryNode) {
+      this.restoreDefaultNode();
+    }
+
+    // When default summary is used, update it.
+    if (this.node === this.defaultNode) {
+      this.updateDefaultNode(this.node);
+    }
+  }
+
+  /**
+   * Override method inherited from `SlotChildObserveController`
+   * to restore and observe the default summary element.
+   *
+   * @protected
+   * @override
+   */
+  restoreDefaultNode() {
+    const { summary } = this;
+
+    // Restore the default summary.
+    if (summary && summary.trim() !== '') {
+      const node = this.attachDefaultNode();
+      this.initNode(node);
```

Comment:
Some tests fail without it. I will check it out, it shouldn't be needed indeed.

Reply:
Refactored logic to move initializer code to `initAddedNode` which is called for both default and custom node.
This makes it unnecessary to call `initNode` here, because `SlotController` calls `initAddedNode` correctly.

Reply:
Removed initializer from controllers and instead updated components to use `slot-content-changed` - this ensures that default and custom summary is handled correctly. Also, the tooltip target is now updated accordingly.